### PR TITLE
Feat: ORV2-1928 applications in progress pagination

### PR DESCRIPTION
**Pull request by erikataot**
# Description
- Modify the `Applications in Progress` tab to use the updated BE call supporting pagination

Screenshot 2024-02-02 153211

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1143-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1143-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1143-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1143-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1143-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1143-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1143-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1143-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/features/permits/hooks/hooks.ts [hunk 1]
```diff
@@ -347,20 +344,3 @@ export const useAmendmentApplicationQuery = (originalPermitId?: string) => {
     refetchOnWindowFocus: false, // prevent unnecessary multiple queries on page showing up in foreground
   });
 };
-
-/**
- * A custom react query hook that fetches applications in progress.
- * @returns List of applications in progress
- */
-export const useApplicationsInProgressQuery = () => {
-  const applicationsInProgressQuery = useQuery({
```
Comment (zgong-gov) [inline frontend/src/features/permits/hooks/hooks.ts:356]: Why did you remove this hook?

File: frontend/src/features/permits/components/permit-list/InProgressApplicationList.tsx [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+  });
+  const [sorting, setSorting] = useState<MRT_SortingState>([
+    {
+      id: "startDate",
+      desc: true,
+    },
+  ]);
+
+  const applicationsQuery = useQuery({
+    queryKey: [
```
Comment (zgong-gov) [inline frontend/src/features/permits/components/permit-list/InProgressApplicationList.tsx]: Shouldn't we wrap this with the hook?

**Comment by zgong-gov** [comment]
The count chip for number of in progress applications is missing (should be '1' here).

onroutebc-missing-in-progress-count

File: frontend/src/features/permits/components/permit-list/ApplicationInProgressColumnDefinition.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+export const ApplicationInProgressColumnDefinition: MRT_ColumnDef<ApplicationInProgress>[] =
+  [
+    {
+      accessorKey: "applicationNumber",
+      header: "Application #",
+      accessorFn: (row) => row.applicationNumber,
+      Cell: (props: { cell: any; row: any }) => {
+        const permitIdStr = `${props.row.original.permitId}`;
+        return (
+          <Link
```
Comment (zgong-gov) [inline frontend/src/features/permits/components/permit-list/ApplicationInProgressColumnDefinition.tsx]: Don't use Link, use CustomNavLink

File: frontend/src/features/permits/hooks/hooks.ts (hunk 1 above)
Comment (erikataot) [reply frontend/src/features/permits/hooks/hooks.ts:356]: because we're not using a hook in the other paginated permit queries. I was trying to follow a similar pattern as the `Active Permits` and the `Expired Permits` tabs

**Comment by erikataot** [comment]
> The count chip for number of in progress applications is missing (should be '1' here).
> 
> onroutebc-missing-in-progress-count

Hmm thanks I'll take a look

File: frontend/src/features/permits/components/dashboard/PermitLists.tsx [hunk 4]
```diff
@@ -1,50 +1,15 @@
...
-        return <Navigate to={ERROR_ROUTES.UNAUTHORIZED} />;
-      }
-      return <ErrorFallback error={error.message} />;
-    }
-  }
-
   const tabs = [
     {
       label: "Applications in Progress",
-      count: applicationsInProgress?.length,
```
Comment (zgong-gov) [inline frontend/src/features/permits/components/dashboard/PermitLists.tsx:41]: We need this to display the number in the count chip for in progress applications tab

File: frontend/src/features/permits/hooks/hooks.ts (hunk 1 above)
Comment (zgong-gov) [reply frontend/src/features/permits/hooks/hooks.ts:356]: does using a hook impact the implementation/logic of the pagination vs not using a hook?

Comment (zgong-gov) [reply frontend/src/features/permits/hooks/hooks.ts:356]: because if it doesn't make a difference, then I feel like we should use the hook(s) for all the various tabs

Comment (zgong-gov) [reply frontend/src/features/permits/hooks/hooks.ts:356]: ok I guess we can leave it like this for now, but in the future if there's additional logic associated with querying we should probably move it into a hook instead.

File: frontend/src/features/permits/components/permit-list/BasePermitList.tsx [hunk 5]
```diff
@@ -31,8 +31,10 @@ import { ERROR_ROUTES } from "../../../../routes/constants";
  */
 export const BasePermitList = ({
   isExpired = false,
+  //isInProgress = false,
```
Comment (krishnan-aot) [inline frontend/src/features/permits/components/permit-list/BasePermitList.tsx]: can be removed?

**Review by krishnan-aot — APPROVED** [review]

File: frontend/src/features/permits/components/permit-list/BasePermitList.tsx (hunk 5 above)
Comment (erikataot) [reply frontend/src/features/permits/components/permit-list/BasePermitList.tsx]: yup good call out. Done.

**Review by zgong-gov — APPROVED** [review]
